### ISSUE TITLE
wlan0 doesn't get IP after successful connect API call

### ISSUE THREAD

**Issue by cecchisandrone** [cjimti/iotwifi#4]
* **I'm submitting a ...**
  - [X] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

* **What is the current behavior?**
After calling connect API call with successful result, wlan0 doesn't get an IP address and traffic can't flow to the gateway.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** 

1. Start iotwifi
2. Call connect API to one of existing networks
3. wlan0 doesn't receive proper IP configuration

* **What is the expected behavior?**
wlan0 should have an IP address in order to communicate with gateway.

* **Please tell us about your environment:**
  
  - Device:
  - OS:

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)
Logs: https://gist.github.com/cecchisandrone/ad6f7348d2a7eab8096e3492d6ab4733

```
wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::ba27:ebff:fee0:496f  prefixlen 64  scopeid 0x20<link>
        ether b8:27:eb:e0:49:6f  txqueuelen 1000  (Ethernet)
        RX packets 72  bytes 4462 (4.3 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 17  bytes 2720 (2.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Comment by cjimti** [comment]
@cecchisandrone this could be something as simple dhcp not running properly on the host. I'll write up some debug steps tonight or tomorrow. I have also not tested on a Pi Zero W, but I have one so I'll test it.

**Comment by cecchisandrone** [comment]
Thanks. Do you think there are some prerequisites on the host on the networking configuration apart from disabling wpa_supplicant?

**Comment by cecchisandrone** [comment]
@cjimti running sudo dhclient wlan0 -v makes it work:
```
sudo dhclient wlan0 -v
Internet Systems Consortium DHCP Client 4.3.5
Copyright 2004-2016 Internet Systems Consortium.
All rights reserved.
For info, please visit https://www.isc.org/software/dhcp/

Listening on LPF/wlan0/b8:27:eb:e0:49:6f
Sending on   LPF/wlan0/b8:27:eb:e0:49:6f
Sending on   Socket/fallback
DHCPDISCOVER on wlan0 to 255.255.255.255 port 67 interval 6
DHCPREQUEST of 192.168.1.116 on wlan0 to 255.255.255.255 port 67
DHCPOFFER of 192.168.1.116 from 192.168.1.3
DHCPACK of 192.168.1.116 from 192.168.1.3
bound to 192.168.1.116 -- renewal in 2985 seconds.
```
I searched into the code but there is no `dhclient` call. Should it be done automatically by wpa_supplicant?

**Comment by cjimti** [comment]
There is no call since this was originally for the Pi 3 which as wired ethernet and hdcp is already handling eth0 by default. Sound like this could be another feature request for the next version which I plan to support more operating environments and devices. It's good to hear that you are connected!

**Comment by cecchisandrone** [comment]
What about routes changes needed?

**Comment by cjimti** [comment]
The device itself should have no DNS issues as that would be resolved with the DHCP, however, routing from the uap0 to wlan0 is going to need some work to do it properly with DNS. The current use case is to only allow wifi network configuration and not act as a router/wifi repeater, but I do see demand for that, it's just not a great solution specifically for the Pi since the onboard wifi is not the most robust. I can see it being useful on devices with multiple higher-end wifi interfaces, so it's something I want to explore. 

The mainline use case is plugging in a device (like Nest or Amazon Alexa) and using your phone to bring it online, then disconnecting.

**Comment by cecchisandrone** [comment]
Yes, I agree about performances, but imagine scenarios in which you deploy a IoT network, where Rpi is used as router / gateway for smart devices or sensors. This is very useful because you test everything abstracting the connectivity and everything can work seamlessly in every network you are (hopefully :)).
